modules/4337/contracts/experimental/P256Verifier.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.0;
+
+import {FCL_ecdsa} from "../vendor/FCL/FCL_ecdsa.sol";
+
+/**
+ * @title P256VerifierWithFallback
+ * @dev A contract that implements a P256 elliptic curve signature verification following the precompile's interface.
+ * The contract provides a fallback function that takes a specific input format and returns a result indicating
+ * whether the signature is valid or not.
+ * The input format is as follows:
+ * - input[  0: 32] = signed data hash
+ * - input[ 32: 64] = signature r
+ * - input[ 64: 96] = signature s
+ * - input[ 96:128] = public key x
+ * - input[128:160] = public key y
+ * The result is a bytes array where the first 32 bytes represent 0x00..00 (invalid) or 0x00..01 (valid).
+ * For more details, refer to the EIP-7212 specification: https://eips.ethereum.org/EIPS/eip-7212
+ */
+contract P256VerifierWithFallback {
+    fallback(bytes calldata input) external returns (bytes memory) {
```

Comment:
You can add an:

```solidity
    function ecdsaVerify(
        bytes32 message,
        uint256 r,
        uint256 s,
        uint256 x,
        uint256 y
    ) internal view abstract returns (bool);
```

And then you could:

```solidity
contract FCLWebAuthnVerifier is WebAuthnVerifier {
    function ecdsaVerify(
        bytes32 message,
        uint256 r,
        uint256 s,
        uint256 x,
        uint256 y
    ) internal view override returns (bool) {
        return FCL_ecdsa.ecdsa_verify(message, r, s, x, y);
    }
}

contract PrecompileWebAuthnVerifier is WebAuthnVerifier {
    function ecdsaVerify(
        bytes32 message,
        uint256 r,
        uint256 s,
        uint256 x,
        uint256 y
    ) internal view override returns (bool) {
        // implement with precompile...
    }
}
```

Reply:
Could you please elaborate on the hierarchy you have in mind? I _think_ I got it, but the message and the file you left a comment on confuse me.